fdbclient/ClientStatusReport.cpp

```diff
@@ -0,0 +1,174 @@
+			healthy = false;
+		} else {
+			reportCoordinators();
+			reportClientInfo();
+			reportStorageServers();
+			reportConnections();
+			statusObj["Healthy"] = healthy;
+		}
+		return StringRef(json_spirit::write_string(json_spirit::mValue(statusObj)));
+	}
+
+private:
+	void reportCoordinators() {
+		auto& connRecord = cx.connectionRecord->get();
+		ClusterConnectionString cs = connRecord->getConnectionString();
+		json_spirit::mArray coordArray;
+		for (const auto& hostName : cs.hostnames) {
+			coordArray.push_back(hostName.toString());
+		}
+		for (const auto& addr : cs.coords) {
+			coordArray.push_back(addr.toString());
+			serverAddresses.insert(addr);
+		}
+		statusObj["Coordinators"] = coordArray;
+		if (cx.coordinator->get().present()) {
+			statusObj["CurrentCoordinator"] = cx.coordinator->get().get().getAddressString();
+		}
+
+		// Update health status
+		if (cs.hostnames.size() + cs.coords.size() == 0) {
+			healthy = false;
+		}
+		if (!cx.coordinator->get().present()) {
+			healthy = false;
+		}
+	}
+
+	void reportClientInfo() {
+		auto& clientInfo = cx.clientInfo->get();
+		statusObj["ClusterID"] = clientInfo.clusterId.toString();
+		json_spirit::mArray grvProxyArr;
+		for (const auto& grvProxy : clientInfo.grvProxies) {
+			serverAddresses.insert(grvProxy.address());
+			grvProxyArr.push_back(grvProxy.address().toString());
+		}
+		statusObj["GrvProxies"] = grvProxyArr;
+		json_spirit::mArray commitProxyArr;
+		for (const auto& commitProxy : clientInfo.commitProxies) {
+			serverAddresses.insert(commitProxy.address());
+			commitProxyArr.push_back(commitProxy.address().toString());
+		}
+		statusObj["CommitProxies"] = commitProxyArr;
+
+		// Update health status
+		if (clientInfo.grvProxies.size() == 0 || clientInfo.commitProxies.size() == 0) {
+			healthy = false;
+		}
+	}
+
+	void reportStorageServers() {
+		json_spirit::mArray storageServerArr;
+		for (const auto& [ssid, serverInfo] : cx.server_interf) {
+			json_spirit::mObject serverDescr;
+			serverDescr["SSID"] = ssid.toString();
+			serverDescr["Address"] = serverInfo->interf.address().toString();
+			serverAddresses.insert(serverInfo->interf.address());
+			storageServerArr.push_back(serverDescr);
+		}
+		statusObj["StorageServers"] = storageServerArr;
+	}
+
+	void reportConnections() {
+		json_spirit::mArray connectionArr;
+		for (const auto& addr : serverAddresses) {
+			connectionArr.push_back(connectionStatusReport(addr));
+		}
+		statusObj["Connections"] = connectionArr;
+		statusObj["NumConnectionsFailed"] = numConnectionsFailed;
+
+		// Update health status
+		if (numConnectionsFailed > 0) {
+			healthy = false;
+		}
+	}
+
+	json_spirit::mObject connectionStatusReport(const NetworkAddress& address) {
+		json_spirit::mObject connStatus;
+		connStatus["Address"] = address.toString();
+
+		auto& peers = FlowTransport::transport().getAllPeers();
+		auto peerIter = peers.find(address);
+
+		FlowTransport::transport().healthMonitor()->getRecentClosedPeers();
```

Comment:
This returns a set of recently closed addresses.

Reply:
That's a left over, need to remove it. Thanks for noticing